### PR TITLE
adding a local registry type that runs outside minikube

### PR DESCRIPTION
**Pull request by sbawaska**

### CODE REVIEW

File: clusters/minikube/start.sh [hunk 1]
```diff
@@ -18,7 +18,7 @@ sudo minikube start --memory=8192 --cpus=4 \
   --vm-driver=${vm_driver} \
   --bootstrapper=kubeadm \
   --extra-config=apiserver.enable-admission-plugins="LimitRanger,NamespaceExists,NamespaceLifecycle,ResourceQuota,ServiceAccount,DefaultStorageClass,MutatingAdmissionWebhook" \
-  --insecure-registry registry.kube-system.svc.cluster.local
+  --insecure-registry $IMAGE_REPOSITORY_HOST
```
Comment (scothis) [inline clusters/minikube/start.sh]: You're introducing a new requirement within the FATS API here. I'd urge you to avoid doing so unless absolutely necessary. In that case, it needs to be documented in the readme and applied to every other cluster/registry.

File: registries/local/start.sh [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+# Enable local registry
+echo "Installing a local registry"
+docker run -d -p 5000:5000 registry:2
+
+dev_ip=172.16.1.1
+sudo su -c "echo \"\" >> /etc/hosts"
+sudo su -c "echo \"$dev_ip       $IMAGE_REPOSITORY_HOST\" >> /etc/hosts"
+sudo ifconfig lo:0 $dev_ip
+kubectl create namespace riff
```
Comment (scothis) [inline registries/local/start.sh]: Let's avoid calling things by the name of a different component in the system that it is not related to.

File: registries/minikube/configure.sh [hunk 3]
```diff
@@ -5,7 +5,8 @@ sudo su -c "echo '{ \"insecure-registries\" : [ \"registry.kube-system.svc.clust
 sudo systemctl daemon-reload
 sudo systemctl restart docker
 
-IMAGE_REPOSITORY_PREFIX="registry.kube-system.svc.cluster.local/u"
+IMAGE_REPOSITORY_HOST="registry.kube-system.svc.cluster.local"
+IMAGE_REPOSITORY_PREFIX="$IMAGE_REPOSITORY_HOST/u"
```
Comment (scothis) [inline registries/minikube/configure.sh]: The `/u` is a turd left over from when `riff namespace init` had a `--registry-user` flag that could not be empty. We don't need it anymore. I'll open an issue to remove it from other registries.

File: clusters/minikube/start.sh (hunk 1 above)
Comment (sbawaska) [reply clusters/minikube/start.sh]: I can create a `kube-system` namespace for local docker repository. I just afraid that it would create confusion if someone was trying to debug.

**Comment by sbawaska** [comment]
@scothis tests finally pass with this new registry type, and I have incorporated feedback, please review.

File: azure-pipelines.yml [hunk 4]
```diff
@@ -9,6 +9,10 @@ jobs:
         imageName: ubuntu-16.04
         cluster: minikube
         registry: dockerhub
+      localregistry:
```
Comment (scothis) [inline azure-pipelines.yml]:
```suggestion
      daemonregistry:
```

File: registries/docker-daemon/cleanup.sh [hunk 5]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+
+# nothing to do
```
Comment (scothis) [inline registries/docker-daemon/cleanup.sh]: This should minimally stop/delete the registry process running in the daemon. Ideally, it would also remove the k8s service and endpoint.

File: registries/docker-daemon/start.sh [hunk 6]
```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+# Enable local registry
+echo "Installing a local registry"
+docker run -d -p 5000:5000 -p 80:5000 registry:2
```
Comment (scothis) [inline registries/docker-daemon/start.sh]: Do we need both port defined? If we switch the k8s service and endpoint to use 80, can we drop 5000?

File: registries/docker-daemon/start.sh [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+
+dev_ip=172.16.1.1
+sudo su -c "echo \"\" >> /etc/hosts"
+sudo su -c "echo \"$dev_ip       registry.kube-system.svc.cluster.local\" >> /etc/hosts"
+sudo ifconfig lo:0 $dev_ip
+cat <<EOF | kubectl create -f -
```
Comment (scothis) [inline registries/docker-daemon/start.sh]:
```suggestion

cat <<EOF | kubectl create -f -
```

File: registries/docker-daemon/start.sh [hunk 8]
```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+# Enable local registry
+echo "Installing a local registry"
```
Comment (scothis) [inline registries/docker-daemon/start.sh]:
```suggestion
echo "Installing a daemon registry"
```

File: registries/minikube/start.sh [hunk 9]
```diff
@@ -1,7 +1,7 @@
 #!/bin/bash
 
 # Enable local registry
-echo "Installing a local registry"
+echo "Installing a local registry inside minikube"
```
Comment (scothis) [inline registries/minikube/start.sh]:
```suggestion
echo "Installing the minikube registry"
```

File: registries/docker-daemon/start.sh (hunk 6 above)
Comment (sbawaska) [reply registries/docker-daemon/start.sh]: I vaguely remember seeing a problem listening on port 80. Will try again.

**Comment by sbawaska** [comment]
@scothis the pipeline failed due to trouble reaching pks. Also, this PR does not impact that part of the pipeline. Should this be merged?

**Comment by scothis** [comment]
@sbawaska the healdsburg environment no longer exists. Try rebasing/merging master
